### DISCUSSION TITLE
Unhandled Promise Rejection - Connection Reset

### DISCUSSION THREAD

**Discussion by inbytesdevteam** [supabase/supabase#719] [Questions]
We're experiencing a 502 Gateway error intermittently on our site for `select` calls. We currently use Supabase with Nuxt.js and Vercel.

Here's the error it brings out:
```
"stack":["Runtime.UnhandledPromiseRejection: FetchError: request to {SUPABASE_LINK_HERE} failed, reason: Client network socket disconnected before secure TLS connection was established","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:326:22)","    at process.EventEmitter.emit (domain.js:483:12)","    at processEmit [as emit] (/var/task/node_modules/signal-exit/index.js:161:32)","    at processPromiseRejections (internal/process/promises.js:209:33)","    at processTicksAndRejections (internal/process/task_queues.js:98:32)"]
```

I was wondering if anyone has experienced this error before? Here are the sample queries on one of the pages that error out:
```
 const { body: posts, error: postsError } = await supabase
      .from('posts')
      .select('*')

 const { body: comments, error: commentsError } = await supabase
      .from('comments')
      .select('*')
```

**Comment by inian** [comment]
Since the error message says Client network socket disconnected, it looks like the client (Vercel) is closing the connection before the TLS handshake finishes. One behaviour we have noticed is that Vercel terminates the function as soon as you respond to the client request. Any in-flight requests after you send the response will not complete.

**Reply by inbytesdevteam** [reply]
@inian ah okay! Have you experienced this as well? Do you know of any workaround?

**Reply by kiwicopple** [reply]
@inbytesdevteam could you please share the full function code that you're using? Usually something is "returning" from the function before all the other requests have completed (so the serverless function "cuts off" all pending request)

**Reply by inian** [reply]
@inbytesdevteam usually we get past it by `await`ing all async requests before returning the response to the user. We can help debug if you can share the function code.

**Reply by inbytesdevteam** [reply]
@kiwicopple @inian thanks for the insights! Here's the function code. Every request has `await` too.

```
 const { body: episodes, error: episodesError } = await supabase
  .from('episodes')
  .select('*')
  .limit(4)
if (episodesError) console.error(episodesError.message)

const {
  body: featuredEvent,
  error: featuredEventError,
} = await supabase
  .from('events')
  .eq('is_featured', 'true')
  .limit(1)
if (featuredEventError) console.error(featuredEventError.message)

const { body: events, error: eventsError } = await supabase
  .from('events')
  .select('*')
  .limit(4)
if (eventsError) console.error(eventsError.message)
```

 We have also noticed that this appears more often after updating the database.

**Reply by inbytesdevteam** [reply]
Hey, @kiwicopple @inian I was wondering if you were able to take a look at the function code above? The error is still randomly popping up.
